### PR TITLE
Refactor Some Application Settings to Configuration File

### PR DESCRIPTION
**Pull request by TCPrimedPaul**
Moved the banner and the azure key vault service setting to the configuration file.
Also removed a few lines of code duplication in Startup.cs.
For your reference, I learned about .NET Core applicaton configuration here: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1

### CODE REVIEW

File: CDTS Blazor/Components/Banners/Banner.razor [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿@inherits BannerBase
 @using CDNApplication.Components.Banners
 
-<section class="banner-container @this.ClassName">
+<section hidden="@(!this.ShowBanner)" class="banner-container @this.ClassName">
```
Comment (AlexandreSanscartier) [inline CDTS Blazor/Components/Banners/Banner.razor:4]: Interesting change just need to be aware of this: [The 'hidden' Attribute is Visibly Weak](https://css-tricks.com/the-hidden-attribute-is-visibly-weak/)
TLDR; Any changes to the display setting in CSS will make this visible again. If we really want to use the hidden attribute we should probably add this somewhere in our CSS file. I'm not a big fan of the !important, but in this case, it's an appropriate work around.
` [hidden] { display: none !important; }`

File: CDTS Blazor/appsettings.json [hunk 2]
```diff
@@ -1,15 +1,32 @@
 {
+  "AllowedHosts": "*",
+  "AzurePublicUrlEndpoints": {
+    "KeyVaultService": "https://kv-seafarer-dev.vault.azure.net/"
+  },
+  "Banners": {
+    "AlphaBanner": {
+      "ShowBanner": "true",
+      "BannerMessage": "This version can change at any moment."
```
Comment (AlexandreSanscartier) [inline CDTS Blazor/appsettings.json:9]: I like the idea of moving the configuration of the banner into its own file, but is it possible to use RESX files for the component instead of having the message in the JSON file?

File: CDTS Blazor/PageHeader.razor [hunk 3]
```diff
@@ -1,2 +1,28 @@
...
+    private BannerOptions betaBannerOptions = new BannerOptions();
+    private BannerOptions warningBannerOptions = new BannerOptions();
+
+    protected override void OnInitialized()
+    {
+        Configuration.GetSection(BannerOptions.AlphaBanner).Bind(alphaBannerOptions);
+        Configuration.GetSection(BannerOptions.BetaBanner).Bind(betaBannerOptions);
+        Configuration.GetSection(BannerOptions.WarningBanner).Bind(warningBannerOptions);
+    }
+}
```
Comment (AlexandreSanscartier) [inline CDTS Blazor/PageHeader.razor:28]: I like the change here. Just one thought I want to put out there is if we grow the number of banners that show up this file will grow as well. This might not be a problem just something to consider if ever we start having 5+ different banner types.

File: CDTS Blazor/Components/Banners/Banner.razor (hunk 1 above)
Comment (TCPrimedPaul) [reply CDTS Blazor/Components/Banners/Banner.razor:4]: I am weak in CSS and thought this was a universal attribute. Definitely open to making it better.

File: CDTS Blazor/appsettings.json (hunk 2 above)
Comment (TCPrimedPaul) [reply CDTS Blazor/appsettings.json:9]: Good point, I can look into that. Plan B could be to list a bilingual message.

File: CDTS Blazor/Components/Banners/Banner.razor (hunk 1 above)
Comment (AlexandreSanscartier) [reply CDTS Blazor/Components/Banners/Banner.razor:4]: In this case it's most likely alright as we shouldn't overwrite the property in our CSS it was more of a here is this interesting fact about this attribute.

File: CDTS Blazor/PageHeader.razor (hunk 3 above)
Comment (TCPrimedPaul) [reply CDTS Blazor/PageHeader.razor:28]: Yah, I just listed the banners I thought we would use eventually (alpha/beta/warning). Even if there was more banners I prefer controlling them from config than having to change code every deployment.

Comment (tc-tibo) [reply CDTS Blazor/PageHeader.razor:28]: Maybe in the future we keep them for Dev and Acc as well so there's no uncertainty for the user with regards to which version they're using.

**Comment by tc-tibo** [comment]
I like the idea of moving things to config files. I think this will allow us to easily set up publish profiles, if I'm not mistaken.

File: CDTS Blazor/PageHeader.razor (hunk 3 above)
Comment (TCPrimedPaul) [reply CDTS Blazor/PageHeader.razor:28]: Ya, solution currently only has appSettings.Development but when we move to acc/prod, we can create new app setting files for those and overload. As far as I understand, it works very much like the old MVC web configs. [Use Multiple Environments in .NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/environments?view=aspnetcore-3.1)

File: CDTS Blazor/appsettings.json (hunk 2 above)
Comment (TCPrimedPaul) [reply CDTS Blazor/appsettings.json]: I translated the banner messages. I prefer it this way so that all banner configurations are at the same place.
